samples/go/ipfs-chat/lib/termui.go

```diff
@@ -0,0 +1,368 @@
+// Copyright 2017 Attic Labs, Inc. All rights reserved.
+// Licensed under the Apache License, version 2.0:
+// http://www.apache.org/licenses/LICENSE-2.0
+
+package lib
+
+import (
+	"fmt"
+	"regexp"
+	"runtime"
+	"strings"
+
+	"github.com/attic-labs/noms/go/d"
+	"github.com/attic-labs/noms/go/datas"
+	"github.com/attic-labs/noms/go/types"
+	"github.com/attic-labs/noms/go/util/math"
+	"github.com/attic-labs/noms/samples/go/ipfs-chat/dbg"
+	"github.com/jroimartin/gocui"
+)
+
+const (
+	allViews    = ""
+	usersView   = "users"
+	messageView = "messages"
+	inputView   = "input"
+
+	searchPrefix = "/s"
+	quitPrefix   = "/q"
+)
+
+type TermUI struct {
+	Gui      *gocui.Gui
+	InSearch bool
+}
+
+var (
+	viewNames   = []string{usersView, messageView, inputView}
+	firstLayout = true
+)
+
+func CreateTermUI(events chan ChatEvent) TermUI {
```

Comment:
Don't you need to return `*TermUI`? Otherwise `TermUI.InSearch` will always be by value.